File: 2015/day-10/day10.py
#! python



def run(values: str, runs=50):
    run = values
    for j in range(runs):
        nextRun = ""
        v = 1
        lastChar = run[0]
        for i in range(1, len(run)):
            c = run[i]
            if c == lastChar:
                v += 1
            else:
                nextRun += f"{v}{lastChar}"
                v = 1
            lastChar = c
        nextRun += f"{v}{c}"
        run = nextRun

        if j == 39:
            p1 = len(run)
        if j == 49:
            p2 = len(run)

    return (p1, p2)


if __name__ == '__main__':

    p1, p2 = run("1113122113")

    print(f"Part 1: {p1}\nPart 2: {p2}")

File: CSharp/aoc/year_2015/day10/day10.cs
namespace Year2015.Day10;


public class Day10
{

    private static string makeNextIter(string currRun)
    {
        StringBuilder nextRun = new();
        int v = 1;
        char currChar = currRun[0];
        char lastChar = currRun[0];
        for (int j = 1; j < currRun.Length; j++)
        {
            currChar = currRun[j];
            if (currChar == lastChar) v++;
            else
            {
                nextRun.Append($"{v}{lastChar}");
                v = 1;
            }
            lastChar = currChar;
        }
        nextRun.Append($"{v}{currChar}");
        return nextRun.ToString();
    }

    public static void Run(string value, int runs = 50)
    {
        int p1 = 0;
        int p2 = 0;
        string run = value;

        for (int i = 0; i < runs; i++)
        {
            run = makeNextIter(run);

            if (i == 39) p1 = run.Length;
            if (i == 49) p2 = run.Length;
        }
        Console.WriteLine($"Part 1: {p1}\nPart 2: {p2}");
    }

}